libs/ts-rest/core/src/lib/client.spec.ts

```diff
@@ -528,16 +560,26 @@ const customClient = initClient(router, {
 type CustomClientGetPostsType = Expect<
   Equal<
     Parameters<typeof customClient.posts.getPosts>[0],
-    | {
-        query?: {
-          take?: number;
-          skip?: number;
-          order?: string;
-        };
-        headers?: Record<string, string>;
-        uploadProgress?: (progress: number) => void;
-      }
-    | undefined
+    {
+      query?: {
+        take?: number;
+        skip?: number;
+        order?: string;
+      };
+      headers: {
+        'x-pagination'?: string;
+        'x-test'?: string;
+        'base-header'?: string;
+        'x-api-key': string;
+      };
+      extraHeaders?: {
```

Comment:
what are `extraHeaders`?

Reply:
Is this to allow the developer to pass extra headers during runtime? 
How is this handled on the express/nestjs side?

Reply:
Yes, exactly! I forgot to mention that in the comments. On the server they will just be merged into the `req.headers`/`params.headers`. It might make sense to separate them on the server too, I'm not sure.